var/spack/repos/builtin/packages/grads/package.py

```diff
@@ -15,35 +15,66 @@ class Grads(AutotoolsPackage):
+    version("2.2.2", sha256="1b5a600d4d407ffcf2fbbbba42037a6e1ebfdb8246ba56b93c628e3c472b4ded")
     version("2.2.1", sha256="695e2066d7d131720d598bac0beb61ac3ae5578240a5437401dc0ffbbe516206")
 
     variant("geotiff", default=True, description="Enable GeoTIFF support")
     variant("shapefile", default=True, description="Enable Shapefile support")
+    variant("grib2", default=True, description="Enable GRIB2 support")
+    variant("dap", default=False, description="Enable DAP support")
+
+    # TODO: This variant depends on the "simple X" library, which is no longer available
+    # from any trusted source. Revisit if this changes.
+    # variant("gui", default=False, description="Enable graphical user interface")
 
-    """
-    # FIXME: Fails with undeclared functions (tdefi, tdef, ...) in gauser.c
-    variant('hdf5', default=False, description="Enable HDF5 support")
-    variant('hdf4', default=False, description="Enable HDF4 support")
-    variant('netcdf', default=False, description="Enable NetCDF support")
-    depends_on('hdf5', when='+hdf5')
-    depends_on('hdf', when='+hdf4')
-    depends_on('netcdf-c', when='+netcdf')
-    """
+    # These variants are broken in 2.2.1
+    # See https://github.com/j-m-adams/GrADS/issues/2
+    variant("hdf5", default=True, when="@2.2.2:", description="Enable HDF5 support")
+    variant("hdf4", default=True, when="@2.2.2:", description="Enable HDF4 support")
+    variant("netcdf", default=True, when="@2.2.2:", description="Enable NetCDF support")
 
+    depends_on("hdf5@:1.10", when="+hdf5")
+    depends_on("hdf", when="+hdf4")
+    depends_on("netcdf-c", when="+netcdf")
+    depends_on("g2c", when="+grib2")
     depends_on("libgeotiff", when="+geotiff")
     depends_on("shapelib", when="+shapefile")
+    depends_on("gadap", when="+dap")
     depends_on("udunits")
     depends_on("libgd")
     depends_on("libxmu")
     depends_on("cairo +X +pdf +fc +ft")
     depends_on("readline")
     depends_on("pkgconfig", type="build")
 
+    # The project is hosted on GitHub for versions 2.2.2 and later
+    def url_for_version(self, version):
+        if version >= Version("2.2.2"):
+            url = "https://github.com/j-m-adams/GrADS/archive/refs/tags/v{}.tar.gz"
+            return url.format(version)
+        else:
+            url = "ftp://cola.gmu.edu/grads/{}/grads-{}-src.tar.gz"
+            return url.format(version.up_to(2), version)
+
+    # Name of grib2 C library has changed in recent versions
+    with when("+grib2"):
+
+        def patch(self):
+            filter_file("grib2c", "g2c", "configure")
+
     def setup_build_environment(self, env):
         env.set("SUPPLIBS", "/")
 
+        # Recent versions configure scripts break without PKG_CONFIG set
+        env.set("PKG_CONFIG", self.spec["pkgconfig"].prefix.bin.join("pkg-config"))
+
+        if self.spec.satisfies("+hdf4") and self.spec["hdf"].satisfies("~shared"):
```

Comment:
Recommend using `in` versus `satisfies` for checking boolean variants.  See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#boolean-variants.

Reply:
@tldahlgren Thanks for the suggestion - I've pushed another commit that makes this change.